scripts/git_changes_utils_test.py

```diff
@@ -0,0 +1,567 @@
+        get_branch_swap = self.swap(
+            common, 'get_current_branch_name', mock_get_branch)
+        with get_branch_swap:
+            self.assertEqual(
+                git_changes_utils.get_changed_files([], 'remote'), {})
+
+    def test_get_changed_files_with_non_empty_ref_list(self) -> None:
+        def mock_get_branch() -> str:
+            return 'branch-1'
+        def mock_compare_to_remote(
+            unused_remote: str,
+            unused_local_branch: str,
+            remote_branch: Optional[str] = None  # pylint: disable=unused-argument
+        ) -> List[str]:
+            return ['A:file1', 'M:file2']
+        def mock_extract_acmrt_files_from_diff(
+            unused_file_diffs: List[git_changes_utils.FileDiff]
+        ) -> List[str]:
+            return ['file1', 'file2']
+
+        get_branch_swap = self.swap(
+            common, 'get_current_branch_name', mock_get_branch)
+        compare_to_remote_swap = self.swap(
+            git_changes_utils, 'compare_to_remote', mock_compare_to_remote)
+        extract_files_swap = self.swap(
+            git_changes_utils, 'extract_acmrt_files_from_diff',
+            mock_extract_acmrt_files_from_diff)
+
+        with compare_to_remote_swap, extract_files_swap, get_branch_swap:
+            self.assertEqual(
+                git_changes_utils.get_changed_files([
+                    git_changes_utils.GitRef(
+                        local_ref='refs/heads/branch1', local_sha1='sha1',
+                        remote_ref='remote/ref1', remote_sha1='rsha1'),
+                    git_changes_utils.GitRef(
+                        local_ref='refs/branch2', local_sha1='sha2',
+                        remote_ref='remote/ref2', remote_sha1='rsha2')
+                    ], 'remote'),
+                {'branch1': (['A:file1', 'M:file2'], ['file1', 'file2'])})
+
+    def test_get_staged_acmrt_files(self) -> None:
+        def mock_git_diff_name_status() -> List[git_changes_utils.FileDiff]:
+            return [
+                git_changes_utils.FileDiff(status=b'M', name=b'file1'),
+                git_changes_utils.FileDiff(status=b'A', name=b'file2')
+            ]
+        git_diff_swap = self.swap(
+            git_changes_utils, 'git_diff_name_status',
+            mock_git_diff_name_status)
+        with git_diff_swap:
+            self.assertEqual(
+                git_changes_utils.get_staged_acmrt_files(),
+                [b'file1', b'file2'])
+
+    def test_get_staged_acmrt_files_with_no_files(self) -> None:
+        def mock_git_diff_name_status() -> List[git_changes_utils.FileDiff]:
+            return []
+        git_diff_swap = self.swap(
+            git_changes_utils, 'git_diff_name_status',
+            mock_git_diff_name_status)
+        with git_diff_swap:
+            self.assertEqual(git_changes_utils.get_staged_acmrt_files(), [])
+
+    def test_get_refs_with_stdin(self) -> None:
+        temp_stdin_file = tempfile.NamedTemporaryFile().name
```

Comment:
Please use a `with` statement to make sure the temporary file gets cleaned up. See https://docs.python.org/3/library/tempfile.html#examples

Reply:
Done.